components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImplTest.java

```diff
@@ -0,0 +1,360 @@
+        endpointRegistryEntryNew.setDefinitionType("WSDL1");
+        endpointRegistryEntryNew.setEndpointDefinition(new ByteArrayInputStream(new byte[]{}));
+
+        Mockito.when(apiMgtDAO.getEndpointRegistryEntryByUUID(endpointRegistryEntryOld.getEntryId()))
+                .thenReturn(endpointRegistryEntryOld);
+        Mockito.when(apiMgtDAO.isRegistryEntryNameExists(endpointRegistryEntryNew))
+                .thenReturn(false);
+
+        endpointRegistry.updateEndpointRegistryEntry(endpointRegistryEntryNew);
+    }
+
+    @Test(expected = APIMgtResourceAlreadyExistsException.class)
+    public void updateEndpointRegistryEntry_existingEntryName() throws APIManagementException {
+        EndpointRegistryEntry endpointRegistryEntryOld = new EndpointRegistryEntry();
+        endpointRegistryEntryOld.setEntryId("abc1");
+        endpointRegistryEntryOld.setName("Entry 1");
+        endpointRegistryEntryOld.setMetaData("{mutualTLS:true}");
+        endpointRegistryEntryOld.setRegistryId(1);
+        endpointRegistryEntryOld.setServiceURL("https://xyz.com");
+        endpointRegistryEntryOld.setServiceType("REST");
+        endpointRegistryEntryOld.setServiceCategory("UTILITY");
+        endpointRegistryEntryOld.setDefinitionURL("https://petstore.swagger.io/v2/swagger.json");
+        endpointRegistryEntryOld.setDefinitionType("OAS");
+        endpointRegistryEntryOld.setEndpointDefinition(new ByteArrayInputStream(new byte[]{}));
+
+        EndpointRegistryEntry endpointRegistryEntryNew = new EndpointRegistryEntry();
+        endpointRegistryEntryNew.setEntryId("abc1");
+        endpointRegistryEntryNew.setName("Entry 2");
+        endpointRegistryEntryNew.setMetaData("{mutualTLS:flase}");
+        endpointRegistryEntryNew.setRegistryId(1);
+        endpointRegistryEntryNew.setServiceURL("https://xyz2.com");
+        endpointRegistryEntryNew.setServiceType("REST");
+        endpointRegistryEntryNew.setServiceCategory("UTILITY");
+        endpointRegistryEntryNew.setDefinitionURL("https://petstore.swagger.io/v2/swagger2.json");
+        endpointRegistryEntryNew.setDefinitionType("WSDL1");
+        endpointRegistryEntryNew.setEndpointDefinition(new ByteArrayInputStream(new byte[]{}));
+
+        Mockito.when(apiMgtDAO.getEndpointRegistryEntryByUUID(endpointRegistryEntryOld.getEntryId()))
+                .thenReturn(endpointRegistryEntryOld);
+        Mockito.when(apiMgtDAO.isRegistryEntryNameExists(endpointRegistryEntryNew))
+                .thenReturn(true);
+
+        endpointRegistry.updateEndpointRegistryEntry(endpointRegistryEntryNew);
+    }
+
+    @Test
+    public void deleteEndpointRegistryEntry() throws APIManagementException {
+        endpointRegistry.deleteEndpointRegistryEntry("entry1");
+    }
+
+    private void compareRegistryInfo(EndpointRegistryInfo expected, EndpointRegistryInfo actual) {
+        org.junit.Assert.assertEquals(expected.getUuid(), actual.getUuid());
```

Comment:
why fully qualified name here?

Reply:
updated in https://github.com/wso2/carbon-apimgt/pull/8534/commits/8aad6c2009f4840cedef91797b02feb0a640fa4e